[CI] fix test_distributed for python 3.8+

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#36542 [CI] fix test_distributed for python 3.8+**

Python 3.8 set the default multiprocessing start mode to spawn for mac (https://docs.python.org/3/library/multiprocessing.html), but we
need fork in these tests, otherwise there are some pickling issues (ex: when attempting to pickle the tempfiles). See more details in https://github.com/pytorch/pytorch/issues/36515
Test: Ensure that these tests succeed when run with python 3.8

Differential Revision: [D21007753](https://our.internmc.facebook.com/intern/diff/D21007753/)

rohan-varma committed:
[CI] fix test_distributed for python 3.8+

Python 3.8 set the default multiprocessing start mode to spawn, but we
need fork in these tests, otherwise there are some pickling issues.
Test: Ensure that these tests succeed when run with python 3.8

Differential Revision: [D21007753](https://our.internmc.facebook.com/intern/diff/D21007753/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[CI] fix test_distributed for python 3.8+"


Python 3.8 set the default multiprocessing start mode to spawn, but we
need fork in these tests, otherwise there are some pickling issues.
Test: Ensure that these tests succeed when run with python 3.8

Differential Revision: [D21007753](https://our.internmc.facebook.com/intern/diff/D21007753/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
Thanks for fixing! in the future, we need to make  test_distributed.py also use MultiProcessingTestCase

rohan-varma committed:
Update on "[CI] fix test_distributed for python 3.8+"


Python 3.8 set the default multiprocessing start mode to spawn for mac (https://docs.python.org/3/library/multiprocessing.html), but we
need fork in these tests, otherwise there are some pickling issues (ex: when attempting to pickle the tempfiles). See more details in https://github.com/pytorch/pytorch/issues/36515
Test: Ensure that these tests succeed when run with python 3.8

Differential Revision: [D21007753](https://our.internmc.facebook.com/intern/diff/D21007753/)

[ghstack-poisoned]
